[IMP] Market Connectors: eBay Marketplace edits (#6214)
odoo/documentation: content/applications/sales/sales/ebay_connector/setup.rst

  invoices created from eBay orders with incoming eBay money.

- Generate a marketplace account deletion/closure notification token. Navigate to
  :menuselection:`Sales app --> Settings --> eBay` to start. Change the mode to
  :guilabel:`Production`, and input random text values for the :guilabel:`Production App Key` and
  for the :guilabel:`Production Cert Key`. Then click on :guilabel:`Generate Token` under the
  :guilabel:`eBay Marketplace Account Deletion/Closure Notifications` section. This token will be
  used during the setup on eBay for the deletion/closure notifications configuration.

.. image:: setup/generate-token.png
   :align: center
   :alt: Generate a verification token in Odoo.

Set up on eBay
==============

Set up eBay developer account
-----------------------------

To start, create an eBay developer account via `eBay's developer portal
<https://go.developer.ebay.com/>`_. This site requires a different login and password than the eBay
account, though the same email address can be used to register. The verification to create a
developer account is around 24 hours.

Set up eBay keyset
------------------

> Once the eBay developer account is created, an application will need to be set up. To set up an
> application, start on `eBay's developer portal <https://go.developer.ebay.com/>`_ and navigate to

Reviewer: ```suggestion
Once the eBay developer account is created, set up an application on `eBay's developer portal
<https://go.developer.ebay.com/>`_. Next, navigate to
```

> the `Hi [username]` at top right of screen, then click on :guilabel:`Application Keysets`. A screen
> will appear with the prompt to :guilabel:`Enter Application Title`. This title can be no more than

Reviewer: suggestion for brevity
```suggestion
the `Hi [username]` at top right of screen, then click :guilabel:`Application Keysets` to open a pop-up that prompts the user to
:guilabel:`Enter Application Title`. This title can be no more than
```

fifty characters. Enter the application name, then choose the appropriate environment (Sandbox or
Production) to generate the first keyset. This application name will not be saved until the keyset
is generated. Click on :guilabel:`Create a keyset` to generate the keyset.

.. warning::

>    Once a Production Keyset is created, the keyset will be disabled. By default, the keyset will
>    been disabled, they can be activated by subscribing to the eBay Marketplace account deletion or
>    closure notifications or by applying to eBay for an exemption. On enabling, the database can make
>    5000 calls per day with this keyset.

Reviewer: This is a rewrite for brevity and clarity based on my understanding of this paragraph. Feel free to accept only parts of the suggestion if it's not really correct (:
```suggestion
   The newly created *production keyset* is disabled by default. Activate it by subscribing to the
   eBay Marketplace 'account deletion or closure notifications' or by applying to eBay for an
   exemption. Upon enabling, the database can make 5000 calls per day using this keyset.
```

   the `Hi [username]` at top right of screen, then :guilabel:`User Access Tokens` and click on
   :guilabel:`Sign in to Sandbox`. The :guilabel:`User Token` can also be accessed by
   clicking on :guilabel:`User Tokens` from the :menuselection:`Application Keys` page.

Test the setup is correct by saving the credentials in Odoo. Once the initial setup is complete a
new menu tab in products will appear called `eBay` with the option to :guilabel:`Sell on eBay`.
See the :doc:`manage` documentation on how to list products.

.. tip::
   Sync Product Categories by clicking on :guilabel:`Product Categories`. Once the categories are
   synced a new menu item should appear called `eBay Category` with eBay categories brought over
   into the database. These are available when listing an eBay item through Odoo.

   .. important::
      If Product Categories beyond 4 paths are required, users will need to manually add those
      paths. This has historically been done by getting a list of all product categories beyond 4
      paths, manually importing them into the Product Category model in Odoo, then linking them
      individually to the product.

> .. seealso::
>    Now that the setup is complete, proceed to either :doc:`create listings <manage>` or :doc:`link
>    existing listings <linking_listings>`.

Reviewer: what you currently have totally works! I just wanted to suggest another option, which is creating a custom admonition block:
```suggestion
.. admonition:: Read more
   Now that the setup is complete, proceed to either:
   - :doc:`create listings <manage>`
   - :doc:`link existing listings <linking_listings>`.
```

.. image:: setup/ebay-variation.png
  :align: center
  :alt: An example on eBay of the variations that can be added to a product.

**Item specifics**, located at the bottom of the listing, detail product-specific information. These
specifics don't sync with Odoo fields by default; a development is required to link these fields.

.. image:: setup/item-specifics.png
  :align: center
  :alt: Item specifics listed on an eBay product.

**Sandbox** and **Production** are terms that are used to categorize the eBay environments as either
still in development/testing (**Sandbox**) OR for use in the the real-instance of the database with
real customer information/dataset (**Production**). It is recommended to start first in the
**Sandbox** to test, and then following the processes below, create a **Production** instance.

.. tip::
   eBay's sandbox environment can be accessed by navigating to `eBay's sandbox portal
   <https://sandbox.ebay.com/>`_ at `https://sandbox.ebay.com/`. eBay's production environment can
   be accessed by navigating to `eBay.com portal <https://www.ebay.com/>`_ or
   `https://www.ebay.com/`.

.. important::

>    The environment selection should remain the same for all environment settings on eBay and on

Author: Yes this is correct!

    - Set up one “base” product (noted in the :guilabel:`Component` field of the :abbr:`BoM (Bill
      of Materials)`) from which all eBay listings will pull from. This will be a storable product
      so stock can be kept. Highlighted in green below, this product will be included in the kit on
      each subsequent “linked” product below.

    - Set up 2+ “linked” products (noted in the :guilabel:`Product` field of the :abbr:`BoM (Bill
      of Materials)`, one for each eBay listing. The product type will be determined by the
      company's accounting settings, as explained in the Odoo documentation. Highlighted in yellow
      below, each product should have a :guilabel:`BoM type` equal to :guilabel:`Kit` and have the
      “base” product as a :guilabel:`component` of the kit. When this linked eBay product is sold,
      the delivery order created will have the base product listed in lieu of the linked product.

    .. image:: setup/products-odoo.png
      :align: center
      :alt: Setting up bill of materials with base product and linked products.

   .. seealso::
      :doc:`../../../inventory_and_mrp/manufacturing/management/bill_configuration/`

>       :doc:`../../../`

Author: Removing this linkage.

     - Cert Key/ID
   * - eBay
     - Dev ID
     - User Token
     - App ID (Client ID)
     - Cert ID (Client Secret)
   * - Odoo
     - Developer Key
     - Production/Sandbox Token
     - Production/Sandbox App Key
     - Production/Sandbox Cert Key

.. important::
   The :guilabel:`Application Keyset` can be accessed by going to `eBay's developer portal
   <https://go.developer.ebay.com/>`_ and navigate to the `Hi [username]` at top right of screen,
   then click on :guilabel:`Application Keysets`. Get to the *User Token* in eBay by navigating to
   the `Hi [username]` at top right of screen, then :guilabel:`User Access Tokens` and click on
   :guilabel:`Sign in to Sandbox`. The :guilabel:`User Token` can also be accessed by
   clicking on :guilabel:`User Tokens` from the :menuselection:`Application Keys` page.

Test the setup is correct by saving the credentials in Odoo. Once the initial setup is complete a
new menu tab in products will appear called `eBay` with the option to :guilabel:`Sell on eBay`.
See the :doc:`manage` documentation on how to list products.

.. tip::
   Sync product categories by clicking :guilabel:`Product Categories`. After syncing, a new menu

>    item, `eBay Category`, appears (from which menu?). These eBay categories are imported from the

Author: Removed and added in clarification

  :menuselection:`Sales app --> Configuration --> Settings`. Under the :guilabel:`eBay` heading,
  change the mode to :guilabel:`Production`, and input random text values for the
  :guilabel:`Production Cert Key`. Then click the :guilabel:`Generate Token` button under the
  :guilabel:`eBay Marketplace Account Deletion/Closure Notifications` section. This token will be
  used during the setup on eBay for the deletion/closure notifications configuration.

.. image:: setup/generate-token.png
   :align: center
   :alt: Generate a verification token in Odoo.

Set up on eBay
==============

Set up eBay developer account
-----------------------------

To start, create an eBay developer account via `eBay's developer portal
<https://go.developer.ebay.com/>`_. This site requires a different login and password than the eBay
account, though the same email address can be used to register. The verification to create a
developer account is around 24 hours.

Set up eBay keyset
------------------

Once the eBay developer account is created, set up an application on `eBay's developer portal
<https://go.developer.ebay.com/>`_. Next, navigate to the :guilabel:`Hi [username]` heading at top

> right of screen, then from the drop-down menu options, click :guilabel:`Application Keysets`.
> Doing so opens a pop-up that prompts the user to :guilabel:`Enter Application Title` (up to fifty

Reviewer 2: ```suggestion
right of screen, then from the drop-down menu options, click :guilabel:`Application Keysets`. Doing
so opens a pop-up that prompts the user to :guilabel:`Enter Application Title` (up to fifty
```

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

To configure notifications or delete the database on a production environment, navigate to the `eBay
developer portal <https://go.developer.ebay.com/>`_. Configure the account deletion/notification
settings in eBay by navigating to the `Hi [username]` at top right of screen, then
:guilabel:`Application Keysets`.

Next, click the :guilabel:`marketplace deletion/account closure notification` option under the
:guilabel:`Production` keyset column. Enter an email under :guilabel:`Email to notify if marketplace
account deletion notification endpoint is down`. Click :guilabel:`Save` to enable the email.

Following this action, enter the :guilabel:`Marketplace account deletion notification endpoint` URL
provided by Odoo. This HTTPs endpoint is found in Odoo by navigating to :menuselection:`Sales app
--> Configuration --> Settings`, in the :guilabel:`eBay Marketplace Account Deletion/Closure
Notifications` field.

Clicking the :guilabel:`Generate Token` button in Odoo below this field creates a verification token
for the eBay production environment. In Odoo, :guilabel:`Copy` the newly created token and navigate

> to eBay to fill in the :guilabel:`Verification token` field. Click :guilabel:`Save` to enable
> the :guilabel:`Event Notification Delivery Method`.

Reviewer 2: ```suggestion
to eBay to fill in the :guilabel:`Verification token` field. Click :guilabel:`Save` to enable the
:guilabel:`Event Notification Delivery Method`.
```

.. image:: setup/account-closure.png
   :align: center
   :alt: Configuring account deletion / notification settings in eBay.

After completing the above fields, click :guilabel:`Send Test Notification` to test the new
notifications. Proceed to the next step when the green check mark appears. Revisit the above
settings if the test post is not as expected.

After configuring notification settings, go back to the :menuselection:`Application Keys` page to
generate production keysets.

Creating the keyset
~~~~~~~~~~~~~~~~~~~

> A successful setup of the notifications enables the ability to create Production Keysets which
> are needed in the remainder of the Odoo configuration. Navigate back to the
> :menuselection:`Application Keys` page generate a production keyset.

Reviewer 2: ```suggestion
A successful setup of the notifications enables the ability to create Production Keysets which are
needed in the remainder of the Odoo configuration. Navigate back to the :menuselection:`Application
Keys` page generate a production keyset.
```

:guilabel:`Time`.

Entering credentials into Odoo
==============================

The previously copied :guilabel:`User Token` and :guilabel:`Application Keyset` are now ready to be
entered into the Odoo database.

Navigate back the eBay settings in Odoo (:menuselection:`Sales app --> Configuration --> Settings
--> eBay`) and paste the following credentials from eBay into the corresponding fields in Odoo.

.. list-table::
   :header-rows: 1
   :stub-columns: 1

   * - Platform
     - Dev Key/ID
     - Token
     - App Key/ID
     - Cert Key/ID
   * - eBay
     - Dev ID
     - User Token
     - App ID (Client ID)
     - Cert ID (Client Secret)
   * - Odoo
     - Developer Key
     - Production/Sandbox Token
     - Production/Sandbox App Key
     - Production/Sandbox Cert Key

.. important::
   The :guilabel:`Application Keyset` can be accessed by going to `eBay's developer portal
   <https://go.developer.ebay.com/>`_ and navigate to the `Hi [username]` at top right of screen,
   then click on :guilabel:`Application Keysets`. Get to the *User Token* in eBay by navigating to
   the `Hi [username]` at top right of screen, then :guilabel:`User Access Tokens` and click on

>    :guilabel:`Sign in to Sandbox`. The :guilabel:`User Token` can also be accessed by
>    clicking on :guilabel:`User Tokens` from the :menuselection:`Application Keys` page.

Reviewer 2: ```suggestion
   :guilabel:`Sign in to Sandbox`. The :guilabel:`User Token` can also be accessed by clicking on
   :guilabel:`User Tokens` from the :menuselection:`Application Keys` page.
```
